src/frontend/mame/ui/swlist.cpp

```diff
@@ -210,16 +186,12 @@ menu_software_list::entry_info *menu_software_list::append_software_entry(const
 	if (entry_updated)
 	{
 		// find the end of the list
-		entryptr = &m_entrylist;
-		while ((*entryptr != nullptr) && (compare_entries(entry, *entryptr, m_ordered_by_shortname) >= 0))
-			entryptr = &(*entryptr)->next;
+		auto iter = m_entrylist.begin();
+		while (iter != m_entrylist.end() && compare_entries(entry, *iter, m_ordered_by_shortname) >= 0)
+			iter++;
```

Comment:
The pre-increment operator (`++iter`) is marginally more efficient on iterators as it doesn't need to create a temporary holding the old value.